components/script/dom/htmltablecellelement.rs

```diff
@@ -123,6 +128,19 @@ impl HTMLTableCellElementLayoutHelpers for LayoutJS<HTMLTableCellElement> {
                 .unwrap_or(LengthOrPercentageOrAuto::Auto)
         }
     }
+
+    fn get_inherited_cellpadding(&self) -> Option<u32> {
+        unsafe {
+            // Follow chain: Cell -> Row -> Section -> Table
+            self.upcast::<Node>().parent_node_ref()
```

Comment:
I'm still not sure whether this is the right thing to do... This works when it's a perfect table, but Firefox does handle a few more cases we should probably also handle:

http://searchfox.org/mozilla-central/rev/2aa0806c598ec433e431728f5ddd3a6847c1a417/dom/html/HTMLTableCellElement.cpp#46